File: 3005-count-elements-with-maximum-frequency/3005-count-elements-with-maximum-frequency.py
class Solution(object):
    def maxFrequencyElements(self, nums):
        mp = {}
        count, max_freq = 0, 0

        for x in nums:
            if x in mp:
                mp[x] += 1
            else:
                mp[x] = 1

            if mp[x] > max_freq:
                max_freq = mp[x]
                count = 0
            
            if mp[x] == max_freq:
                count += 1
        
        return count * max_freq
        

File: 3005-count-elements-with-maximum-frequency/3005-count-elements-with-maximum-frequency.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        int count = 0, maxFreq = 0;

        for(int it : nums) {
            mp.put(it, mp.getOrDefault(it, 0) + 1);

            if(mp.get(it) > maxFreq) {
                maxFreq = mp.get(it);
                count = 0;
            }

            if(mp.get(it) == maxFreq) count++;
        }

        return count * maxFreq;
    }
}